### ISSUE TITLE
exists(pattern.exact()) at the java version not really exact....

### ISSUE THREAD

**Issue by quilombodigital** [sikuli/sikuli#19]
Hi,

The java pattern.exact() suffers from a problem, it does not always match correctly the image at the screen.
The Sikuli IDE works normally with the same image.
I used similar(0.995f) to get the score and I noticed the match score returns numbers like 0.9999996423721313.
I tried other images.
I guess the problem is related with how python and java round the score number before checking the exists..

[]s, ricardo

**Comment by RaiMan** [comment]
Not really clear, what you are complaining.

What you give with similar(0.995f) is the minimum value that found matches should have.
This means a match that returns a score of 0.9999996423721313 is correct, since this is true:
0.9999996423721313 >= 0.995f

**Comment by quilombodigital** [comment]
what I was complaining is that by using exact(), the match was not working with the exists() method.
I grabbed the image directly from the screen, so it was a perfect match, and it was not working.
I used similar() just to discover what was the score I was having, in this case  0.9999996423721313, not 1.0f
I tried more than one image with the same effect.

Since this works at the ide, and not at the java version, I thought that the python exists() probably round the result 2 or 3 digits before comparing, but this is just a guess. :)

**Comment by RaiMan** [comment]
exact() (meaning similar(1.0)) does not work - known problem.
use similar(0.99) instead if you want exact matches.

**Comment by quilombodigital** [comment]
yep, but I found another issue, the screen I am checking has a "disabled state", that means it makes the screen grayscale when it is disabled.
I want to find the image in its colored state, but not in its grayscale state.
the problem is that similar(0.99) still finds the grayscale version of the image (that I dont want)
Well. I did a workaround that finds another images, but I´m just leaving here registered that I could not find a simple way to differentiate colored and grayscale exact matches.
